### PR TITLE
Replace bosco_findplatform use of deprecated platform.distro()

### PR DESCRIPTION
**Pull request by brianhlin**
(HTCONDOR-242)

Parse /etc/os-release parsing instead

Note that the new `findversion()` constructs the Bosco distro/version return strings instead of using the defined variables (https://github.com/htcondor/htcondor/blob/master/src/condor_contrib/bosco/bosco_findplatform#L9-L20). I'll submit another PR getting rid of those along with some other PEP 8 fixes.

### CODE REVIEW

**Comment by brianhlin** [comment]
I decided against fixing this up for PEP 8 and a separate PR. It's not worth the chance of introducing a whitespace bug.

File: src/condor_contrib/bosco/bosco_findplatform [hunk 1]
```diff
@@ -53,135 +52,84 @@ def findversion_mac(detail=False):
...
+    try:
+      distro_info = get_distro_info()
+    except FileNotFoundError:
+      return "UNKNOWN"
+
+    distro = distro_info['ID']
+    bosco_distro = DISTRO_MAPPING[distro]
+    major_ver = distro_info['VERSION_ID'].split('.')[0]
+
+    if distro == 'linuxmunt':
```
Comment (matyasselmeci) [inline src/condor_contrib/bosco/bosco_findplatform:111]:
```suggestion
    if distro == 'linuxmint':
```

File: src/condor_contrib/bosco/bosco_findplatform [hunk 2]
```diff
@@ -53,135 +52,84 @@ def findversion_mac(detail=False):
...
-  return UNKNOWN
+      return "UNSUPPORTED"
+
+    try:
+      distro_info = get_distro_info()
+    except FileNotFoundError:
+      return "UNKNOWN"
+
+    distro = distro_info['ID']
+    bosco_distro = DISTRO_MAPPING[distro]
```
Comment (matyasselmeci) [inline src/condor_contrib/bosco/bosco_findplatform:108]:
```suggestion
    try:
        bosco_distro = DISTRO_MAPPING[distro]
    except KeyError:
        return "UNKNOWN"
```

File: src/condor_contrib/bosco/bosco_findplatform [hunk 3]
```diff
@@ -53,135 +52,84 @@ def findversion_mac(detail=False):
...
+        return "UNKNOWN"
+
+    return f'{bosco_distro}{major_ver}'
+
+  return "UNKNOWN"
 
 def findversion_detail():
   #if not os.name=='posix':
-  #  return UNKNOWN 
+  #  return "UNKNOWN"
```
Comment (matyasselmeci) [inline src/condor_contrib/bosco/bosco_findplatform:123]: just delete these

**Review by matyasselmeci — APPROVED** [review]
One change needed but pre-approving.

File: src/condor_contrib/bosco/bosco_findplatform (hunk 1 above)
Comment (brianhlin) [reply src/condor_contrib/bosco/bosco_findplatform:111]: You haven't heard of the popular distro Linux Munt?

**Review by timtheisen — APPROVED** [review]
This looks good. Although, I would have just used the the Ubuntu code name rather than make a special table for Linux Mint.

**Pull request merged**